docs/design/speed-test.md

```diff
@@ -0,0 +1,210 @@
+# Upload Speed Test for Backup Storage Location (Object Storage)
+
+## Abstract
+
+This document presents the design of a Custom Resource Definition (CRD) and its controller to test the upload speed from an OpenShift cluster to cloud object storage.
+The design leverages the BackupStorageLocation configuration from the OADP operator’s DPA CRD to specify object storage locations for testing. The controller will use
```

Comment:
we intend to only measure BSLs created by DPA and not all BSLs in OADP namespace?

Reply:
So these are 2 separate things:
- You can independently create UST CR for any BSL config (so this would work for OADP BSL and Non-OADP BSLs)
- We have a trigger for OADP BSLs (in this case DPA controller will create a UST CR)
So all the scenarios are covered I believe.